src/reality-capture.ts

```diff
@@ -0,0 +1,273 @@
+     * @param {EngineVersion} version The reconstruction engine version.
+     * @param {string[]} metadataname Fine-tuning parameter field names indexed to match the metadata_value index, beginning from index 0.
+     * @param {string[]} metadatavalue Fine-tuning parameter values indexed to match the metadata_name index, beginning from index 0.
+     * @returns {Promise<IPhotoScene>} A JSON object containing details of the photoscene that was created, with property 'photosceneid' ID of the photoscene that was created.
+     * @throws Error when the request fails, for example, due to invalid request.
+     */
+    async createPhotoScene(scenename: string, callback: string, format: OutputFormat, scenetype: SceneType, gpstype?: GpsType, hubprojectid?: string, hubfolderid?: string, version?: EngineVersion, metadataname?: string[], metadatavalue?: string[]): Promise<IPhotoScene> {
+        const params: any = {
+            scenename,
+            callback,
+            format,
+            scenetype
+        }
+        if (scenetype === 'aerial' && gpstype) { 
+            params.gpstype = gpstype;
+        }
+        if (hubprojectid && hubfolderid) {
+            params.hubprojectid=hubprojectid;
+            params.hubfolderid=hubfolderid;
+        }
+        if (version) {
+            params.version=version;
+        }
+        if (
+            scenetype === 'aerial'
+            && metadataname
+            && metadatavalue
+            && metadataname.length > 0
+            && metadatavalue.length > 0
+            && metadataname.length === metadatavalue.length
+        ) {
+            for (let i=0; i<metadataname.length; i++) {
+                params[`metadataname[${i}]`]=metadataname[i];
+                params[`metadatavalue[${i}]`]=metadatavalue[i];
+            }
+        }
+        const headers: { [key: string]: string } = {};
+        headers['Content-Type'] = 'application/x-www-form-urlencoded';
+        return this.post('photoscene', querystring.stringify(params), headers, WriteTokenScopes);
+    }
+
+    /**
+     * Adds one or more files to a photoscene.
+     * Only JPEG images are supported.
+     * Maximum number of files in a single request: 20
+     * Maximum size of a single file: 128 MB
+     * Maximum uncompressed size of image in memory: 512 MB
+     * Note: Uploaded files will be deleted after 30 days.
+     * {@link https://forge.autodesk.com/en/docs/reality-capture/v1/reference/http/file-POST|docs}.
+     * @async
+     * @param {string} photosceneid Specifies the ID of the photoscene to add the files to.
+     * @param {FileType} type Specifies the type of file being uploaded: image (default) or survey
+     * @param {string[]} files Specifies the files to be uploaded - these may be URLs (i.e. http://, https://) or local files. For externally stored files, please verify that the URLs are publically accessible.
```

Comment:
This code comment says that the `files` parameter may include _local files_? In that case, the function code should probably read the file's content from the file system and send it with the POST request, right?

Reply:
Petr, for local files the syntax remains file[index], I don't think we need to read the files locally, just specify local file path.

```
curl -v 'https://developer.api.autodesk.com/photo-to-3d/v1/file' \
  -X 'POST' \
  -H 'Authorization: Bearer eyjhbGCIOIjIuzI1NiISimtpZCI6...' \
  -F "photosceneid=hcYJcrnHUsNSPII9glhVe8lRF6lFXs4NHzGqJ3zdWMU" \
  -F "type=image" \
  -F "file[0]=@c:/sample_data/_MG_9026.jpg" \
  -F "file[1]=@c:/sample_data/_MG_9027.jpg"
```

Reply:
The `@<filepath>` syntax is a special functionality of _curl_. The `@` symbol will make _curl_ load the file from that file path and include its content in the request. So I'm afraid in our case the `files` parameter will need to be of type `string[] | Buffer[]`, and we will need to make sure that we include the file contents in the right way...

Reply:
According to the [docs](https://forge.autodesk.com/en/docs/reality-capture/v1/reference/http/file-POST/#headers) the request for adding pictures to a photo scene can be either `application/x-www-form-urlencoded` (when you're only submitting URLs) or `multipart/form-data` (when you're submitting actual files). Note that we already use a `form-data` dependency to upload files in DA: https://github.com/petrbroz/forge-server-utils/blob/41d14b7b11045fac54747c362e91f107d1cd17b2/src/design-automation.ts#L392-L413.

Reply:
So here's what I would suggest:
- add a method called `addImageURLs(photosceneid: string, type: FileType, urls: string[])` that will only submit URLs of pictures and not local files); that's basically the method you have today
- add another method (now or perhaps later) called `addImages(photosceneid: string, type: FileType, files: Buffer[])` that will upload the actual file content using `FormData` as we do in Design Automation

Reply:
I added two separate functions to account for the two different ways to add images to a photoscene.